### PR TITLE
Syntax widget: Add special values.

### LINKED ISSUE #178: Document global special values (e.g. __MODULE__)
These are pretty fun:

- `__POS_OF__(string)`
- `__LINE__`
- `__LOC__`
- `__FILE__`
- `__LOC_OF__`
- `__MODULE__`
- `__LINE_OF__`
- `__dirname`

Those should probably be part of the `Build System` sidebar category.

### PR DESCRIPTION
**Pull request by fhammerschmidt**
Fixes #178 
I tried to create meaningful examples which showcase how the values/functions work.

### CODE REVIEW

**Comment by fhammerschmidt** [comment]
I found `__dirname ` to not work, though.

File: pages/docs/manual/latest/global-special-values.mdx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+### Position
+
+`__POS__` returns a tuple consisting of `(file_name, line_number, column_number_start, column_number_end)` corresponding to the location in the file at which you put this very expression.
+
+It basically provides the same information as `__LOC__`, but separated into multiple values. 
+
+<CodeTab labels={["ReScript", "JS Output"]}>
+
+```res example
+/* padding comment */ __POS__->Js.log
```
Comment (woeps) [inline pages/docs/manual/latest/global-special-values.mdx:90]: Not sure if it's worth the lengthened example, but for newcomers to the language the example could benefit from using destructuring to emphasize on the tuple:
```suggestion
let (file_name, line_number, column_number_start, column_number_end) = __POS__

file_name->Js.log
line_number->Js.log
column_number_start->Js.log
column_number_end->Js.log
```

The js tab would need to be adapted accordingly.

The same suggestion (desctructuring the tuple) also applys to the following functions returning a tuple.

File: pages/docs/manual/latest/global-special-values.mdx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+</CodeTab>
+
+### Location of
+
+`__LOC_OF__(expression)` returns a tuple consisting of `(location, expression)` corresponding to the location of the expression you pass as a parameter here.
+
+<CodeTab labels={["ReScript", "JS Output"]}>
+
+```res example
+__LOC_OF__(() => None)->Js.log;
```
Comment (woeps) [inline pages/docs/manual/latest/global-special-values.mdx:106]: According to my previous comment, destructuring could help newcomers. Not inlining the function demonstrates a common usage.
```suggestion
let f = () => None
let (loc, f') = __LOC_OF__(f)

loc->Js.log
f'->Js.log
```
The js tab would need to be adapted accordingly.

File: pages/docs/manual/latest/global-special-values.mdx [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+</CodeTab>
+
+### Line of
+
+`__LINE_OF__(expression)` returns a tuple consisting of `(line, expression)`, where line is the line number at which the expression you pass as a parameter here appears in the file.
+
+<CodeTab labels={["ReScript", "JS Output"]}>
+
+```res example
+__LINE_OF__(() => None)->Js.log;
```
Comment (woeps) [inline pages/docs/manual/latest/global-special-values.mdx:122]: same as the previous comment:
```suggestion
let f = () => None
let (line, f') = __LINE_OF__(f)

line->Js.log
f'->Js.log
```

File: pages/docs/manual/latest/global-special-values.mdx [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+```js
+console.log([1, (function (param) {})]);
+```
+
+</CodeTab>
+
+### Position of
+
+`__POS_OF__(expression)` returns a tuple consisting of `(position, expression)`, corresponding to the position of the expression you pass as a parameter here.
```
Comment (woeps) [inline pages/docs/manual/latest/global-special-values.mdx:133]: Information about the structure of `position` was already mentioned, but repeating it here makes sense to me.
```suggestion
`__POS_OF__(expression)` returns a tuple consisting of `(position, expression)`, corresponding to the position of the expression you pass as a parameter here. The `position` is a tuple consistinig of `(file_name, line_number, column_number_start, column_number_end)`.
```

**Review by woeps — COMMENTED** [review]
Thanks for taking the matter of documenting these functions into you hands!

Maybe it would be a good idea to mention that `expression` doesn't need to be a function. - hence the name `expression`

File: pages/docs/manual/latest/global-special-values.mdx (hunk 1 above)
Comment (fhammerschmidt) [reply pages/docs/manual/latest/global-special-values.mdx:90]: Good idea.
I just realized that we should stick to camelCase though.

**Comment by fhammerschmidt** [comment]
Changed everything according to your suggestions. Thanks!

**Comment by fhammerschmidt** [comment]
Ok, as discussed with @ryyppy, I split the section up into cards for the syntax widget instead.

**Comment by ryyppy** [comment]
Awesome. Thanks!

**Pull request merged**